### ISSUE TITLE
Combine "dateFunction": momentDateFormatter" and"customCallback": populateTpl

### ISSUE THREAD

**Issue by saturday1** [jasonmayes/Twitter-Post-Fetcher#77]
I am trying to use the template version (Example 8) combined with moment (Example 6, I think). But then I get problems with the moment js. It does not run the function at all and it does not catch the language `moment.locale('sv_SE');`. The code looks like this:

        <script src="//code.jquery.com/jquery-1.12.0.min.js"></script>
        <script type="text/javascript" src="//cdnjs.cloudflare.com/ajax/libs/moment.js/2.8.3/moment.min.js"></script>
	
        <script type="text/javascript" src="js/twitterFetcher_min.js"></script>
	<script>
	
	moment.locale('sv_SE');
	
	var config = {
		"id": '573452057987973120',
		"dataOnly": true,
		"maxTweets": 3,
		"dateFunction": momentDateFormatter,
		"customCallback": populateTpl
	};
	
	function momentDateFormatter(date, dateString) {
		return moment(dateString).fromNow();
	}
	
	twitterFetcher.fetch(config);
	
	function populateTpl(tweets){
		var html = '<ul>';
		for (var i = 0, lgth = tweets.length; i < lgth ; i++) {
			var tweetObject = tweets[i];
			
			html += '<li>'
			+ (tweetObject.image ? '<div class="tweet-img"><img src="'+tweetObject.image+'" /></div>' : '')
			+ '<p class="tweet-content">' + tweetObject.tweet + '</p>'
			+ '<p class="tweet-infos">Posted on the ' + tweetObject.time + ', by ' + tweetObject.author + '</p>'
			+ '<p class="tweet-link"><a href="' + tweetObject.permalinkURL + '">Link</a></p>'
			+ '</li>';
		}
		html += '</ul>';
		
		$('#twitter-feed').append(html);
	}
	</script>
</body>
</html>

You can see all HTML as well over here: https://cdn.axiell.com/dev/twitter/test/index.html

How do I run moment before the template function is running?

**Comment by saturday1** [comment]
Also, how can I see all possible tweetObjects?

**Comment by jasonmayes** [comment]
I'm afraid I am not a user of moment, you may want to reach out to the guy who added support for moment to ask him how to go about doing that.

**Comment by jasonmayes** [comment]
I can confirm Twitter Fetcher calls the function see this fiddle: https://jsfiddle.net/sxvfeza7/

There must be something not being used right with Moment in that case which is outside the scope of this project.

**Comment by saturday1** [comment]
Yes, above example works as intended, also over here (I have similar solution already in place). It is only when I use "dataOnly" it stops working. See the fiddle here which represents my code: https://jsfiddle.net/sxvfeza7/3/. As you can see it does not write anything in console.log. When I turn "dataonly" to false, it does work with dateFormatter.

**Comment by jasonmayes** [comment]
Ah, I see. I shall have a look at that.

**Comment by cdarlens** [comment]
I want to show on my website my the lastest tweets

These are steps:
- Go to www.twitter.com and sign in as normal, go to your settings page.
- Go to "Widgets" on the left hand side.
- Create a new widget for what you need eg "user time line" or "search" etc.
- Now go back to settings page, and then go back to widgets page and you should see the widget you just created. Click edit.
- Look at the URL in your web browser, you will see a long number like this: “727179718227836928”
- Use this as your ID below instead!


As you can see in the image attached. I dont have the option to create a user timeline widget only search

![twitterfetcherissue](https://cloud.githubusercontent.com/assets/16642768/17040736/92d0aa1e-4f6f-11e6-9212-14971a34eabb.png)

**Comment by jasonmayes** [comment]
Hello there. Please use latest version of Twitter Fetcher which does not
need an ID now.

**Comment by CoalaWeb** [comment]
I'm not sure if it helps but I submitted a pull request for this type of situation in the past as I found the **dataonly** option wasn’t allowing me to format the date. I forked the [Fiddle](https://jsfiddle.net/CoalaWeb/5afpkuzh/1/) @saturday1 created and you can see my changes from line 295-318 (open to improvement)

I can then format the date with out needing to load moment using some like this:

`function dateFormatter(date) {
        var options = {weekday: "short", year: "numeric", month: "short", day: "numeric"};
        return date.toLocaleDateString("sv-SE", options);
    }`

I'm using it with the CMS Joomla and I load in the current language being used so the dates are formated as needed.

Cheers,

**Comment by eballeste** [comment]
when setting `dataOnly` to `true`, you can modify the raw data from the callback function

```javascript
customCallback: (data) => {
  const convertedData = data.map((post) => {
    const { timestamp } = post;
    const unixTimestamp = new Date(timestamp).getTime() / 1000;
    post.timestamp = unixTimestamp;
    
    return post;
  });

   // do something with convertedData
}
```
